feat!: extract storenode cycle to go-waku api

richard-ramos opened the pull request:
Requires:
- https://github.com/waku-org/go-waku/pull/1223

Extracts the storenode cycle code to go-waku. This PR probably makes sense to be reviewed along with the one from go-waku ^.  It's mostly ready to be reviewed. I'll change it to draft once go-waku's PR is complete (it is just missing the unit tests).

richard-ramos committed:
feat!: extract storenode cycle to go-waku api

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (100)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_multiplication_x: | a338199a | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/1/) | 2024-09-19 20:19:07 | ~1 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/1//consoleText) |
| :x: | a338199a | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/1/) | 2024-09-19 20:19:09 | ~1 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/1/consoleText) |
| :x: | a338199a | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/1/) | 2024-09-19 20:19:15 | ~1 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/1/consoleText) |
| :x: | a338199a | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5857/1/) | 2024-09-19 20:19:29 | ~1 min | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5857/1/consoleText) |
| :heavy_multiplication_x: | a338199a | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/1/) | 2024-09-19 20:20:03 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/1//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | 11380458 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/2/) | 2024-09-19 20:30:07 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/2//consoleText) |
| :heavy_multiplication_x: | 11380458 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/2/) | 2024-09-19 20:30:42 | ~2 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/2//consoleText) |
| :heavy_check_mark: | 11380458 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5857/2/) | 2024-09-19 20:31:39 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-240919-202840-113804-pr5857.zip) |
| :heavy_check_mark: | 11380458 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/2/) | 2024-09-19 20:33:07 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240919-202835-113804-pr5857.aar) |
| :heavy_check_mark: | 11380458 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/2/) | 2024-09-19 20:33:19 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240919-202836-113804-pr5857.zip) |
| | | | | | | |
| :heavy_check_mark: | f5dc5cb2 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/3/) | 2024-09-20 19:15:51 | ~2 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/3//consoleText) |
| :heavy_check_mark: | f5dc5cb2 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/3/) | 2024-09-20 19:15:53 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240920-191325-f5dc5c-pr5857.aar) |
| :heavy_check_mark: | f5dc5cb2 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5857/3/) | 2024-09-20 19:15:55 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-240920-191329-f5dc5c-pr5857.zip) |
| :heavy_multiplication_x: | f5dc5cb2 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/3/) | 2024-09-20 19:16:04 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/3//consoleText) |
| :heavy_check_mark: | f5dc5cb2 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/3/) | 2024-09-20 19:17:25 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240920-191325-f5dc5c-pr5857.zip) |
| | | | | | | |
| :heavy_check_mark: | 7539682f | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/4/) | 2024-09-20 19:19:32 | ~1 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/4//consoleText) |
| :heavy_multiplication_x: | 7539682f | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/4/) | 2024-09-20 19:19:35 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/4//consoleText) |
| :heavy_check_mark: | 7539682f | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/4/) | 2024-09-20 19:19:55 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240920-191824-753968-pr5857.aar) |
| :heavy_check_mark: | 7539682f | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5857/4/) | 2024-09-20 19:20:22 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-240920-191828-753968-pr5857.zip) |
| :heavy_check_mark: | 7539682f | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/4/) | 2024-09-20 19:21:44 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240920-191825-753968-pr5857.zip) |
| | | | | | | |
| :heavy_multiplication_x: | f4aa80ee | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/5/) | 2024-09-20 19:24:50 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/5//consoleText) |
| :heavy_check_mark: | f4aa80ee | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/5/) | 2024-09-20 19:25:29 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240920-192328-f4aa80-pr5857.aar) |
| :heavy_check_mark: | f4aa80ee | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/5/) | 2024-09-20 19:25:40 | ~2 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/5//consoleText) |
| :heavy_check_mark: | f4aa80ee | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5857/5/) | 2024-09-20 19:25:46 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-240920-192330-f4aa80-pr5857.zip) |
| :heavy_check_mark: | f4aa80ee | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/5/) | 2024-09-20 19:27:17 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240920-192330-f4aa80-pr5857.zip) |
| | | | | | | |
| :heavy_check_mark: | d9d12473 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/6/) | 2024-09-20 19:32:01 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240920-192952-d9d124-pr5857.aar) |
| :heavy_check_mark: | d9d12473 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/6/) | 2024-09-20 19:32:12 | ~2 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/6//consoleText) |
| :heavy_check_mark: | d9d12473 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5857/6/) | 2024-09-20 19:32:21 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-240920-192956-d9d124-pr5857.zip) |
| :heavy_check_mark: | d9d12473 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/6/) | 2024-09-20 19:33:49 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240920-192954-d9d124-pr5857.zip) |
| :heavy_multiplication_x: | d9d12473 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/6/) | 2024-09-20 19:35:53 | ~5 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/6//consoleText) |
| | | | | | | |
| :x: | 9f4b0828 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/7/) | 2024-10-15 20:37:44 | ~1 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/7/consoleText) |
| :x: | 9f4b0828 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/7/) | 2024-10-15 20:38:05 | ~1 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/7/consoleText) |
| :heavy_multiplication_x: | 9f4b0828 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/7/) | 2024-10-15 20:38:14 | ~1 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/7//consoleText) |
| :x: | 9f4b0828 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5857/7/) | 2024-10-15 20:38:22 | ~2 min | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5857/7/consoleText) |
| :heavy_multiplication_x: | 9f4b0828 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/7/) | 2024-10-15 20:39:09 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/7//consoleText) |
| | | | | | | |
| :heavy_check_mark: | ef3f160d | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5857/8/) | 2024-10-15 21:15:05 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241015-211118-ef3f16-pr5857.zip) |
| :heavy_check_mark: | ef3f160d | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/8/) | 2024-10-15 21:15:33 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241015-211117-ef3f16-pr5857.zip) |
| :heavy_check_mark: | ef3f160d | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/8/) | 2024-10-15 21:15:50 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241015-211115-ef3f16-pr5857.aar) |
| :heavy_check_mark: | ef3f160d | [#8](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/8/) | 2024-10-15 21:16:48 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/8//consoleText) |
| :heavy_multiplication_x: | ef3f160d | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/8/) | 2024-10-15 21:19:04 | ~7 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/8//consoleText) |
| | | | | | | |
| :heavy_check_mark: | b69a3eaf | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/9/) | 2024-10-15 21:38:55 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241015-213638-b69a3e-pr5857.aar) |
| :heavy_check_mark: | b69a3eaf | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5857/9/) | 2024-10-15 21:39:25 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241015-213643-b69a3e-pr5857.zip) |
| :heavy_check_mark: | b69a3eaf | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/9/) | 2024-10-15 21:40:08 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241015-213640-b69a3e-pr5857.zip) |
| :heavy_check_mark: | b69a3eaf | [#9](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/9/) | 2024-10-15 21:42:02 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/9//consoleText) |
| :heavy_multiplication_x: | b69a3eaf | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/9/) | 2024-10-15 21:51:12 | ~14 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/9//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 65d7a851 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/10/) | 2024-10-15 22:02:21 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241015-215948-65d7a8-pr5857.aar) |
| :heavy_check_mark: | 65d7a851 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5857/10/) | 2024-10-15 22:02:48 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241015-215952-65d7a8-pr5857.zip) |
| :heavy_check_mark: | 65d7a851 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/10/) | 2024-10-15 22:03:30 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241015-215949-65d7a8-pr5857.zip) |
| :heavy_check_mark: | 65d7a851 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/10/) | 2024-10-15 22:05:08 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/10//consoleText) |
| :heavy_multiplication_x: | 65d7a851 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/10/) | 2024-10-15 22:13:54 | ~13 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/10//consoleText) |
| | | | | | | |
| :heavy_check_mark: | a06eb72c | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/11/) | 2024-10-15 22:25:28 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241015-222308-a06eb7-pr5857.aar) |
| :heavy_check_mark: | a06eb72c | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5857/11/) | 2024-10-15 22:25:55 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241015-222312-a06eb7-pr5857.zip) |
| :heavy_check_mark: | a06eb72c | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/11/) | 2024-10-15 22:26:41 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241015-222310-a06eb7-pr5857.zip) |
| :heavy_check_mark: | a06eb72c | [#11](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/11/) | 2024-10-15 22:28:25 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/11//consoleText) |
| :heavy_check_mark: | a06eb72c | [#11](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/11/) | 2024-10-15 22:52:27 | ~29 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/11//consoleText) |
| | | | | | | |
| :x: | 384310ec | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5857/12/) | 2024-10-21 20:35:59 | ~12 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5857/12/consoleText) |
| :heavy_check_mark: | 384310ec | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/12/) | 2024-10-21 20:40:05 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241021-203545-384310-pr5857.zip) |
| :heavy_check_mark: | 384310ec | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/12/) | 2024-10-21 20:40:38 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241021-203544-384310-pr5857.aar) |
| :heavy_multiplication_x: | 384310ec | [#12](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/12/) | 2024-10-21 20:42:51 | ~6 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/12//consoleText) |
| :heavy_check_mark: | 384310ec | [#12](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/12/) | 2024-10-21 21:06:27 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/12//consoleText) |
| | | | | | | |
| :x: | 1658d4a0 | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5857/13/) | 2024-10-21 20:38:18 | ~11 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5857/13/consoleText) |
| :heavy_check_mark: | 1658d4a0 | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/13/) | 2024-10-21 20:43:27 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241021-204052-1658d4-pr5857.aar) |
| :heavy_check_mark: | 1658d4a0 | [#13](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/13/) | 2024-10-21 20:43:33 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241021-204018-1658d4-pr5857.zip) |
| :x: | 1658d4a0 | [#14](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5857/14/) | 2024-10-21 20:47:43 | ~8.8 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5857/14/consoleText) |
| :heavy_check_mark: | 1658d4a0 | [#13](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/13/) | 2024-10-21 20:48:37 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/13//consoleText) |
| :heavy_multiplication_x: | 1658d4a0 | [#13](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/13/) | 2024-10-21 21:09:42 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/13//consoleText) |
| | | | | | | |
| :x: | 1960b977 | [#15](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5857/15/) | 2024-10-21 21:14:08 | ~11 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5857/15/consoleText) |
| :heavy_check_mark: | 1960b977 | [#14](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/14/) | 2024-10-21 21:16:11 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241021-211354-1960b9-pr5857.aar) |
| :heavy_check_mark: | 1960b977 | [#14](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/14/) | 2024-10-21 21:16:46 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241021-211355-1960b9-pr5857.zip) |
| :heavy_check_mark: | 1960b977 | [#14](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/14/) | 2024-10-21 21:19:39 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/14//consoleText) |
| :heavy_check_mark: | 1960b977 | [#14](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/14/) | 2024-10-21 21:45:07 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/14//consoleText) |
| :x: | 1960b977 | [#16](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5857/16/) | 2024-10-21 21:58:25 | ~10 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5857/16/consoleText) |
| :x: | 1960b977 | [#17](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5857/17/) | 2024-10-21 22:00:08 | ~11 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5857/17/consoleText) |
| :heavy_check_mark: | 1960b977 | [#18](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5857/18/) | 2024-10-21 22:05:22 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241021-220039-1960b9-pr5857.zip) |
| | | | | | | |
| :heavy_check_mark: | c17f9ef7 | [#15](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/15/) | 2024-10-23 20:54:34 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241023-205120-c17f9e-pr5857.zip) |
| :heavy_multiplication_x: | c17f9ef7 | [#15](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/15/) | 2024-10-23 20:55:01 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/15//consoleText) |
| :heavy_check_mark: | c17f9ef7 | [#19](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5857/19/) | 2024-10-23 20:56:29 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241023-205123-c17f9e-pr5857.zip) |
| :heavy_check_mark: | c17f9ef7 | [#15](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/15/) | 2024-10-23 20:57:02 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241023-205119-c17f9e-pr5857.aar) |
| :heavy_multiplication_x: | c17f9ef7 | [#15](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/15/) | 2024-10-23 20:57:43 | ~6 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/15//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 12a0c7cd | [#16](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/16/) | 2024-10-23 21:07:19 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241023-210457-12a0c7-pr5857.aar) |
| :heavy_check_mark: | 12a0c7cd | [#20](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5857/20/) | 2024-10-23 21:07:45 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241023-210500-12a0c7-pr5857.zip) |
| :heavy_check_mark: | 12a0c7cd | [#16](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/16/) | 2024-10-23 21:10:47 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/16//consoleText) |
| :heavy_check_mark: | 12a0c7cd | [#16](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/16/) | 2024-10-23 21:16:01 | ~11 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241023-210459-12a0c7-pr5857.zip) |
| :heavy_check_mark: | 12a0c7cd | [#16](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/16/) | 2024-10-23 21:36:16 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/16//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 62d4ce9b | [#21](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5857/21/) | 2024-11-07 09:10:25 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241107-090602-62d4ce-pr5857.zip) |
| :heavy_check_mark: | 62d4ce9b | [#17](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/17/) | 2024-11-07 09:11:15 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/17//consoleText) |
| :heavy_check_mark: | 62d4ce9b | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-5857/1/) | 2024-11-07 09:11:22 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241107-090607-62d4ce-pr5857.zip) |
| :heavy_check_mark: | 62d4ce9b | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-5857/1/) | 2024-11-07 09:12:17 | ~6 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-241107-090609-62d4ce-pr5857.zip) |
| :heavy_check_mark: | 62d4ce9b | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-5857/1/) | 2024-11-07 09:12:18 | ~6 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241107-090603-62d4ce-pr5857.zip) |
| :heavy_check_mark: | 62d4ce9b | [#17](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/17/) | 2024-11-07 09:13:26 | ~7 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241107-090600-62d4ce-pr5857.aar) |
| :heavy_check_mark: | 62d4ce9b | [#17](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/17/) | 2024-11-07 09:14:01 | ~7 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241107-090602-62d4ce-pr5857.zip) |
| :heavy_check_mark: | 62d4ce9b | [#17](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/17/) | 2024-11-07 09:37:50 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/17//consoleText) |
| | | | | | | |
| :heavy_check_mark: | e9c399b2 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-5857/2/) | 2024-11-22 14:15:56 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241122-141112-e9c399-pr5857.zip) |
| :heavy_check_mark: | e9c399b2 | [#22](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5857/22/) | 2024-11-22 14:16:01 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241122-141110-e9c399-pr5857.zip) |
| :heavy_check_mark: | e9c399b2 | [#18](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/18/) | 2024-11-22 14:16:13 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241122-141109-e9c399-pr5857.zip) |
| :heavy_check_mark: | e9c399b2 | [#18](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/18/) | 2024-11-22 14:16:28 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/18//consoleText) |
| :heavy_check_mark: | e9c399b2 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-5857/2/) | 2024-11-22 14:16:30 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241122-141113-e9c399-pr5857.zip) |
| :heavy_check_mark: | e9c399b2 | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-5857/2/) | 2024-11-22 14:16:36 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-241122-141117-e9c399-pr5857.zip) |
| :heavy_check_mark: | e9c399b2 | [#18](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/18/) | 2024-11-22 14:17:11 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241122-141107-e9c399-pr5857.aar) |
| :heavy_multiplication_x: | e9c399b2 | [#18](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/18/) | 2024-11-22 14:42:06 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/18//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_multiplication_x: | 033a4456 | [#19](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/19/) | 2024-11-22 15:25:35 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/19//consoleText) |
| :heavy_check_mark: | 033a4456 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-5857/3/) | 2024-11-22 15:27:54 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241122-152322-033a44-pr5857.zip) |
| :heavy_multiplication_x: | 033a4456 | [#19](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/19/) | 2024-11-22 15:28:04 | ~4 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/19//consoleText) |
| :heavy_check_mark: | 033a4456 | [#19](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/19/) | 2024-11-22 15:28:18 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241122-152319-033a44-pr5857.zip) |
| :heavy_check_mark: | 033a4456 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-5857/3/) | 2024-11-22 15:28:44 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241122-152323-033a44-pr5857.zip) |
| :heavy_check_mark: | 033a4456 | [#23](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5857/23/) | 2024-11-22 15:28:53 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241122-152321-033a44-pr5857.zip) |
| :heavy_check_mark: | 033a4456 | [#3](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-5857/3/) | 2024-11-22 15:29:06 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-241122-152328-033a44-pr5857.zip) |
| :heavy_check_mark: | 033a4456 | [#19](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/19/) | 2024-11-22 15:29:29 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241122-152318-033a44-pr5857.aar) |
| | | | | | | |
| :heavy_check_mark: | 7959e55d | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-5857/4/) | 2024-11-22 15:32:35 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241122-152758-7959e5-pr5857.zip) |
| :heavy_check_mark: | 7959e55d | [#20](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/20/) | 2024-11-22 15:32:57 | ~4 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5857/20//consoleText) |
| :heavy_check_mark: | 7959e55d | [#4](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-5857/4/) | 2024-11-22 15:33:08 | ~3 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-241122-152909-7959e5-pr5857.zip) |
| :heavy_check_mark: | 7959e55d | [#20](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5857/20/) | 2024-11-22 15:33:29 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241122-152823-7959e5-pr5857.zip) |
| :heavy_check_mark: | 7959e55d | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-5857/4/) | 2024-11-22 15:34:14 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-241122-152849-7959e5-pr5857.zip) |
| :heavy_check_mark: | 7959e55d | [#24](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5857/24/) | 2024-11-22 15:34:28 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241122-152857-7959e5-pr5857.zip) |
| :heavy_check_mark: | 7959e55d | [#20](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5857/20/) | 2024-11-22 15:35:43 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241122-152934-7959e5-pr5857.aar) |
| :heavy_check_mark: | 7959e55d | [#20](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/20/) | 2024-11-22 15:57:21 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5857/20//consoleText) |

richard-ramos committed:
refactor_: decouple go-waku api from go-waku wakunode

igor-sirotin commented on eth-node/bridge/geth/wakuv2.go:
```diff
@@ -338,14 +306,75 @@ func (w *gethWakuV2Wrapper) ConfirmMessageDelivered(hashes []common.Hash) {
+	return w.waku.StorenodeCycle.PerformStorenodeTask(fn, opts...)
+}
+
+func (w *gethWakuV2Wrapper) DisconnectActiveStorenode(ctx context.Context, backoff time.Duration, shouldCycle bool) {
```
Why this custom `shouldCycle` parameter here, maybe it should be in waku as well? 
I don't even see `DisconnectActiveStorenode` used anywhere actually 😄

richard-ramos replied:
It's used in `SetPinnedMailserver` and `ToggleUseMailserver` in `protocol/messenger_mailserver.go`. The `shouldCycle` parameter is used to indicate whether a new mailserver should be chosen or not.

igor-sirotin commented on eth-node/bridge/geth/wakuv2.go:
```diff
@@ -338,14 +306,75 @@ func (w *gethWakuV2Wrapper) ConfirmMessageDelivered(hashes []common.Hash) {
+	}
+
+	criteria := store.FilterCriteria{
+		TimeStart:     proto.Int64(int64(batch.From) * int64(time.Second)),
+		TimeEnd:       proto.Int64(int64(batch.To) * int64(time.Second)),
```
Would be nice to refactor `MailserverBatch` to have `From`/`To` fields as `time.Time` to avoid such weird conversions. Considering that we do it initially as well:
https://github.com/status-im/status-go/blob/a06eb72c012834100feafc1a90b7d950b1d4d57d/protocol/messenger_mailserver.go#L175-L177

And then we have to convert them once again here, when passing to go-waku...

richard-ramos replied:
I did this change, but while working on it, i realized the madness of how we handle time in status-go: we use uint64 for clock values (milliseconds), on waku we use time.Time, on mailserver code and message gaps calculation we use uint32 (in seconds, i imagine this is 'inherited' from back when we used to have this logic on the mobile app).

This lack of consistency is awful. In a subsequent PR i'll attempt to get rid of the uint32 code for mailserver code and gaps to use time.Time, but perhaps we should consider to later also attempt to get rid of the milliseconds time (uint64) and use time.Time consistently across all codebase?

igor-sirotin replied:
> use time.Time consistently across all codebase

💯

igor-sirotin commented on protocol/messenger.go:
```diff
@@ -883,33 +833,36 @@ func (m *Messenger) Start() (*MessengerResponse, error) {
 		go func() {
 			defer gocommon.LogOnPanic()
-			<-available
+			<-m.transport.OnStorenodeAvailableOneShot()
```
We probably need some `<-ctx.Done()` here as well?
Not related to your change, but would be nice to fix on the way.

igor-sirotin replied:
Actually, why not `WaitForAvailableStoreNode` here? 
A function like `OnStorenodeAvailableOneShot` is quite specific. And only used here, but could easily be replaced with `WaitForAvailableStoreNode` and infinite timeout 🤔

igor-sirotin commented on eth-node/bridge/geth/wakuv2.go:
```diff
@@ -338,14 +306,75 @@ func (w *gethWakuV2Wrapper) ConfirmMessageDelivered(hashes []common.Hash) {
+	return w.waku.StorenodeCycle.StorenodeAvailableEmitter.Subscribe()
+}
+
+func (w *gethWakuV2Wrapper) WaitForAvailableStoreNode(timeout time.Duration) bool {
+	return w.waku.StorenodeCycle.WaitForAvailableStoreNode(context.TODO(), timeout)
+}
```
`context` should probably passed as argument to `WaitForAvailableStoreNode`
```suggestion
func (w *gethWakuV2Wrapper) WaitForAvailableStoreNode(ctx context.Context, timeout time.Duration) bool {
	return w.waku.StorenodeCycle.WaitForAvailableStoreNode(ctx, timeout)
}
```

igor-sirotin commented on protocol/messenger_communities.go:
```diff
@@ -5155,3 +5157,28 @@ func (m *Messenger) startRequestMissingCommunityChannelsHRKeysLoop() {
+
+// getCommunityMailserver returns the active mailserver if a communityID is present then it'll return the mailserver
+// for that community if it has a mailserver setup otherwise it'll return the global mailserver
+func (m *Messenger) getCommunityMailserver(communityID ...string) peer.ID {
```
At least for the new functions 😄 
```suggestion
func (m *Messenger) getCommunityStorenode(communityID ...string) peer.ID {
```

igor-sirotin commented on protocol/messenger_communities.go:
```diff
@@ -5155,3 +5157,28 @@ func (m *Messenger) startRequestMissingCommunityChannelsHRKeysLoop() {
+		return m.transport.GetActiveStorenode()
+	}
+
+	peerID, _ := ms.PeerID()
```
Why ignore error here?

richard-ramos replied:
Oops!

igor-sirotin commented on protocol/messenger_mailserver.go:
```diff
@@ -85,73 +82,49 @@ func (m *Messenger) scheduleSyncChat(chat *Chat) (bool, error) {
 
-	return m.findNewMailserver()
-}
+	go func() {
```
```suggestion
	go func() {
	    defer gocommon.LogOnPanic()
```

igor-sirotin replied:
Also, pass some context to `PeformStorenodeTask`?

igor-sirotin commented on protocol/messenger_mailserver.go:
```diff
@@ -85,73 +82,49 @@ func (m *Messenger) scheduleSyncChat(chat *Chat) (bool, error) {
-	if pinnedMailserver == nil {
-		m.disconnectActiveMailserver(graylistBackoff)
-	}
+func (m *Messenger) performStorenodeTask(task func() (*MessengerResponse, error), opts ...history.StorenodeTaskOption) (*MessengerResponse, error) {
```
This wrapper looks quite difficult. We start a goroutine that has a callback that calls a task, whilst receiving result/error through channels 😵

Any chances to simplify this somehow?

richard-ramos replied:
Should be simpler now!

igor-sirotin commented on protocol/messenger_mailserver.go:
```diff
@@ -1172,13 +917,13 @@ func (m *Messenger) fetchMessages(chatID string, duration time.Duration) (uint32
 		}
 
-		m.logger.Debug("fetching messages", zap.String("chatID", chatID), zap.String("mailserver", ms.Name))
+		m.logger.Debug("fetching messages", zap.String("chatID", chatID), zap.Stringer("storenodeID", peerID))
```
Never heard of `storenodeID` before 😄 
```suggestion
		m.logger.Debug("fetching messages", zap.String("chatID", chatID), zap.Stringer("peerID", peerID))
```

igor-sirotin commented on protocol/storenodes/storenodes.go:
```diff
@@ -51,14 +55,17 @@ func (m *CommunityStorenodes) GetStorenodeByCommunityID(communityID string) (mai
-			if snode.StorenodeID == id {
-				return true
+			commStorenodeID, err := utils.GetPeerID(snode.Address)
+			if err == nil {
+				if commStorenodeID == peerID {
+					return true
+				}
```
```suggestion
			if err == nil && commStorenodeID == peerID {
			    return true
```

igor-sirotin commented on services/ext/api.go:
```diff
@@ -1408,10 +1408,6 @@ func (api *PublicAPI) RequestAllHistoricMessagesWithRetries(forceFetchingBackup
 	return api.service.messenger.RequestAllHistoricMessages(forceFetchingBackup, true)
 }
 
-func (api *PublicAPI) DisconnectActiveMailserver() {
-	api.service.messenger.DisconnectActiveMailserver()
-}
-
```
This counts as a breaking change, please mark the commit accordingly

igor-sirotin reviewed (commented):
👏

osmaczko reviewed (approved):
Good job moving the logic to go-waku, it fits well there :+1: 


I believe this PR carries a high risk and should be thoroughly reviewed by QAs from both platforms before merging. I've just requested a review from the QAs. Are there any specific instructions or steps that should be followed to ensure the affected areas are tested properly?

pavloburykh commented:
> Good job moving the logic to go-waku, it fits well there 👍
> 
> I believe this PR carries a high risk and should be thoroughly reviewed by QAs from both platforms before merging. I've just requested a review from the QAs. Are there any specific instructions or steps that should be followed to ensure the affected areas are tested properly?

@osmaczko, thank you for reaching out to the QA team. We're currently preparing to cut the release branch for the upcoming 2.31 version. Since we want to avoid including any risky changes in this release, we will be ready to test this PR once the 2.31 branch is finalized and released.

@richard-ramos meanwhile, we would appreciate if you create a corresponding mobile PR pointing at these changes and providing some details on what areas are affected and should be tested in scope of this PR. Thank you!

richard-ramos commented:
@pavloburykh 
Yeah, I wouldn't even consider testing and merging this PR until after 2.31 is released.
 I'll create PRs for both desktop and mobile once I implement fixes for the observations from the code review

Thank you!

kaichaosun commented on eth-node/types/waku.go:
```diff
@@ -194,12 +196,60 @@ type Waku interface {
+	OnStorenodeChanged() <-chan peer.ID
+
+	// OnStorenodeNotWorking is triggered when the last active storenode fails to return results consistently
+	OnStorenodeNotWorking() <-chan struct{}
```
I'm wondering if it's possible to use a state machine for different status of store node to simplify the channels here, like
```
type State int

const (
    Offline State = iota
    Online
)

// Define StoreNodeState struct
type StoreNodeState struct {
    status State
    peerID peer.ID
}

// Define Waku struct
type Waku struct {
    storeNodeState StoreNodeState
}

// Function to handle state changes
func handleStoreNodeStateChange(newState chan StoreNodeState) {
    for {
        select {
        case state := <-newState:
            if state.status == Online {
                doA()
            }
            if state.status == Offline {
                doB()
            }
        }
    }
}

// Placeholder for actual functions to be called
func doA() {
    fmt.Println("Doing A: Node is online")
}

func doB() {
    fmt.Println("Doing B: Node is offline")
}
```

richard-ramos replied:
This looks like a good idea! Will tackle it as part of https://github.com/waku-org/go-waku/issues/1246

kaichaosun commented on protocol/messenger_communities.go:
```diff
@@ -5155,3 +5157,28 @@ func (m *Messenger) startRequestMissingCommunityChannelsHRKeysLoop() {
+	ms, err := m.communityStorenodes.GetStorenodeByCommunityID(communityID[0])
+	if err != nil {
+		if !errors.Is(err, storenodes.ErrNotFound) {
+			m.logger.Error("getting storenode for community, using global", zap.String("communityID", communityID[0]), zap.Error(err))
```
It seems you are using `m.transport.GetActiveStorenode()` to get the global store node no match with the error is.

richard-ramos replied:
yeah, that seems to be the intention of the original function, IIUC:
https://github.com/status-im/status-go/blob/develop/protocol/messenger_mailserver_cycle.go#L375-L389

richard-ramos committed:
fix!: code review

richard-ramos committed:
fix_: simplify `performStorenodeTask`

richard-ramos commented:
@igor-sirotin i think I managed to implement/fix all of your suggestions and observations. If this PR gets your blessings I'll proceed to create PRs for desktop/mobile QA. Thank you!

igor-sirotin reviewed (approved):
👍

kaichaosun reviewed (approved):
👍

churik commented:
@richard-ramos is it ready to be tested?
can you please specify what areas can be affected (I suppose the whole messaaging, offline-online handling, fetching from store nodes)?

can you please update https://github.com/status-im/status-mobile/pull/21494 accordingly and move PR to `E2E tests` column when it is ready?

Thanks!

richard-ramos commented:
@churik , done! i've just rebased the PR and moved it to the e2e column.
The areas that are affected are those related to retrieving message history for 1:1 messages and communities, as well as confirming that a message has been sent.

pavloburykh commented:
> @churik , done! i've just rebased the PR and moved it to the e2e column. The areas that are affected are those related to retrieving message history for 1:1 messages and communities, as well as confirming that a message has been sent.

Hey @richard-ramos! Thanks for the PR. In case you have missed, I have logged some issues in mobile PR https://github.com/status-im/status-mobile/pull/21494#issuecomment-2462508858 , please take a look once you have a chance.

richard-ramos committed:
fix_: filter criteria

pavloburykh commented:
@richard-ramos tested on mobile in https://github.com/status-im/status-mobile/pull/21494. No issues from my side.
